### PR TITLE
Feature/various minor updates

### PR DESCRIPTION
**Pull request by Derstilon**
Added filtering for predefined materials
Fixed Direction Handeling for Beam
Added World Zone Material parser

### CODE REVIEW

File: converter/shieldhit/geo.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 from math import log10, ceil, isclose
 from scipy.spatial.transform import Rotation as R
 
+PREDEVINED_MATERIALS = [0, 1000]
```
Comment (grzanka) [inline converter/shieldhit/geo.py:7]: Add a meaningful name for 0 and 1000. It is not clear now which one is vacuum and which - black hole

File: converter/shieldhit/parser.py [hunk 2]
```diff
@@ -236,7 +237,7 @@ def _parse_zones(self, json: dict) -> None:
                 # lists are numbered from 0, but shieldhit zones are numbered from 1
                 id=idx+1,
                 figures_operators=self._parse_csg_operations(zone["unionOperations"]),
-                material=self._get_material_id(zone["materialUuid"]),
```
Comment (retkiewi) [inline converter/shieldhit/parser.py:239]: Trailing commas are good, they sometimes eliminate merge conflicts.

File: converter/shieldhit/geo.py [hunk 3]
```diff
@@ -249,5 +251,7 @@ def get_mat_string(self) -> str:
         """Generate mat.dat config."""
         # we increment idx because shieldhit indexes from 1 while python indexes lists from 0
         material_strings = [
-            self.material_template.format(idx=idx+1, mat=mat[1])for idx, mat in enumerate(self.materials)]
+            self.material_template.format(idx=idx+1, mat=mat[1])
+            for idx, mat in enumerate(self.materials)
+            if mat[1] not in PREDEVINED_MATERIALS]
```
Comment (retkiewi) [inline converter/shieldhit/geo.py:256]: You shouldn't need to do that, 0 and 1000 aren't valid ICRU's so they should not be in `self.materials`. Better to blacklist them in `converter.shieldhit.parser.ShieldhitParser._parse_materials`.

File: converter/shieldhit/geo.py (hunk 1 above)
Comment (retkiewi) [inline converter/shieldhit/geo.py:7]:
```suggestion
PREDEFINED_MATERIALS = [0, 1000]
```

File: converter/shieldhit/geo.py (hunk 3 above)
Comment (retkiewi) [inline converter/shieldhit/geo.py:256]:
```suggestion
            if mat[1] not in PREDEFINED_MATERIALS]
```

File: converter/shieldhit/parser.py [hunk 4]
```diff
@@ -246,27 +247,26 @@ def _parse_zones(self, json: dict) -> None:
...
             Zone(
                 uuid="",
                 id=len(self.geo_mat_config.zones)+1,
-                # slightly larger world zone - world zone
+                # world zone defined by bounding figure and contained zones
                 figures_operators=self._calculate_world_zone_operations(len(self.geo_mat_config.figures)),
-                # the last material is the black hole
-                material=VACUUM_MATERIAL
+                # the material of the world zone is usually defined as vacuum
+                material=self._get_material_id(world_zone["materialUuid"])
```
Comment (retkiewi) [inline converter/shieldhit/parser.py:261]: What if it is vacuum? Shouldn't we check if it isn't 1000?

Comment (retkiewi) [reply converter/shieldhit/parser.py:261]:
```suggestion
                material=self._get_material_id(world_zone["materialUuid"]) if not world_zone["materialUuid"] == 1000 else 1000
```

**Review by retkiewi — APPROVED** [review]

File: converter/shieldhit/__init__.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+DEFALUT_MATERIALS = {
```
Comment (grzanka) [inline converter/shieldhit/__init__.py:1]: Rename to DEFAULT_MATERIALS

Comment (grzanka) [inline converter/shieldhit/__init__.py:1]: Replace with Enum:

https://docs.python.org/3/library/enum.html

**Pull request merged**